src/sync/impl/synchronize.js

```diff
@@ -84,6 +84,10 @@ export default async function synchronize({
   }, 'sync-synchronize-apply')
 
   // push phase
+  if (!pushChanges) {
```

Comment:
I think the if statement should be scoped differently so that we get to the final logs (finishedAt, phase=done), which are still worth noting.

Reply:
I missed that part indeed!
Addressed.